docs/development/design/build-images.rst

```diff
@@ -191,94 +299,122 @@ Examples of these versions are:
-
-* ``ubuntu20-base``: ~5Gb
-* ``ubuntu20-py27``: ~150Mb
-* ``ubuntu20-py36``: ~210Mb
-* ``ubuntu20-py39``: ~20Mb
-* ``ubuntu20-conda47``: ~713Mb
+After this design document gets implemented and tested,
+all our current images (``stable``, ``latest``, ``testing``) will be deprecated and their usage will be de-motivated.
+However, we could keep them on our builders to give users a good time to migrate their projects to the new ones.
+New projects shouldn't be able to select these images and they will be forced to use ``build.os`` and ``build.languages``.
 
-which is about ~6Gb in total, we still have plenty of space.
+We may want to keep only the latest Ubuntu LTS releases available in production,
+with a special consideration for our current Ubuntu 18.04 LTS on ``stable``, ``latest`` and ``testing`` because 100% of the projects depend on them currently.
+Once Ubuntu 22.04 LTS is released, we should communicate that Ubuntu 20.04 LTS is deprecated,
+and give users 2 years to migrate to a newer image (1 year less than the official Ubuntu support for LTS images)
 
-We could keep ``stable``, ``latest`` and ``testing`` for some time without worry too much.
-New projects shouldn't be able to select these images and they will be forced to use ``ubuntu20``
-if they don't specify one.
+.. warning::
 
-We may want to keep the two latest Ubuntu LTS releases available in production.
-At the moment of writing this they are:
+   Deleting ``-base`` images from the build servers **will make project's builds to fail**.
+   We want to keep supporting them as much as we can, but having a well-defined deprecation policy is a win.
 
-* Ubuntu 18.04 LTS (our ``stable``, ``latest`` and ``testing`` images)
-* Ubuntu 20.04 LTS (our new ``ubuntu20``)
 
-Once Ubuntu 22.04 LTS is released, we should deprecate Ubuntu 18.04 LTS,
-and give users 6 months to migrate to a newer image.
+Work required and rollout plan
+------------------------------
 
+The following steps are required to support the full proposal of this document.
 
-Work required
--------------
+#. allow users to install extras languages requirements via config file
 
-There are a lot of work to do here.
-However, we want to prioritize it based on users' impact.
+   * update config file to support ``build.os`` and ``build.languages`` config
+   * modify builder code to run ``asdf install`` for all supported languages
 
-#. allow users to install packages with APT
+#. build a new base Docker image with new structure (``ubuntu20-base``)
 
-   * update config file to support ``build.system_packages`` config
-   * modify builder code to run ``apt-get install`` as ``root`` user
+   * build new image with Ubuntu 20.04 LTS and pre-installed ``asdf`` with all its plugins
+   * do not install any language version on base image
+   * deploy builders with new base image
 
-#. allow users to install extras via config file
+#. update builders to install ``build.languages`` selected by the user
```

Comment:
How is this different than `modify builder code to run ``asdf install`` for all supported languages`?

Reply:
This point does exactly what you are asking. Iterates over `build.languages` and call `asdf install` --we need to update the code that runs on builders to do that.

Reply:
So this is basically `deploy the code`?

Reply:
Not 100% sure what you mean here, but I guess so.

I removed this point because it's already mentioned in the first one as: "update config file to support ``build.os`` and ``build.languages`` config" and "modify builder code to run ``asdf install`` for all supported languages"